### PR TITLE
Added plan to record ion chamber dark current

### PR DESCRIPTION
**Pull request by canismarko**
This PR also includes a new shutter device that functions as a positioner. This device has not been tested at the beamline because the controls will not be available until technical commissioning has begun at 25-ID.

### CODE REVIEW

File: environment.yml
[the remark this answers is not in the document]
Comment (canismarko) [reply environment.yml:1]: I think this is fixed if you install ``ophyd-registry >= 1.3.3``

Before that, children of the devices did not get registered, which I think is why you're getting that error.

File: src/firefly/tests/test_grid_scan_window.py
[the remark this answers is not in the document]
Comment (canismarko) [reply src/firefly/tests/test_grid_scan_window.py:1]: Yeah, let me see what I can do about this. One of the side-effects of using ``AsyncSlot``s. The slot is running in the event loop when the test ends, and it has to destroy the task. Maybe I can look up if there are existing tasks and just cancel them before the test runner ends. Or else I could try not running them at all in the tests unless they're awaited in the test.

File: environment.yml
[the remark this answers is not in the document]
Comment (canismarko) [reply environment.yml:1]: Those are two separate warnings. The MotorFlyer will indeed be removed soon, but I don't think it has anything to do with pyarrow. I'll see if the tests pass with pyarrow >= 14.0.1 as the warning suggests.

File: src/haven/plans/auto_gain.py [hunk 1]
```diff
@@ -111,23 +112,24 @@ def next_gain(self, df: pd.DataFrame):
...
     prefer: str = "middle",
     max_count: int = 28,
     queue: Queue = None,
 ):
-    """An adaptive Bluesky plan for optimizing pre-amp gains.
+    """An adaptive Bluesky plan for optimizing ion chamber
+    pre-amp gains.
 
-    For each detector, the plan will search for the range of gains
+    For each ion chambe, the plan will search for the range of gains
```
[the remark this answers is not in the document]
Comment (canismarko) [reply src/haven/plans/auto_gain.py]: Fixed: https://github.com/spc-group/haven/pull/250/commits/c77bc2887c09597ab2edc41cb00a5e18c9a27916

File: src/haven/tests/test_auto_gain_plan.py
[the remark this answers is not in the document]
Comment (Cathyhjj) [reply src/haven/tests/test_auto_gain_plan.py:1]: later ophyd version fixed~

File: src/firefly/tests/test_grid_scan_window.py
[the remark this answers is not in the document]
Comment (canismarko) [reply src/firefly/tests/test_grid_scan_window.py:1]: I added a short timeout to the test_grid_scan_window tests to let the tasks finish. Not elegant, but I think it'll work for now?

File: environment.yml
[the remark this answers is not in the document]
Comment (canismarko) [reply environment.yml:1]: pyarrow version 17 gave me errors, so instead I added the pyarrow-hotfix to the environment. Does installing that package fix the warning for you?

File: src/firefly/tests/test_grid_scan_window.py
[the remark this answers is not in the document]
Comment (Cathyhjj) [reply src/firefly/tests/test_grid_scan_window.py:1]: Thanks! I am missing `haven.instrument.synApps`

[the remark this answers is not in the document]
Comment (Cathyhjj) [reply src/firefly/tests/test_grid_scan_window.py:1]: I think your last commits deleted a few of your modules; not sure whether it's by accident?

[the remark this answers is not in the document]
Comment (canismarko) [reply src/firefly/tests/test_grid_scan_window.py:1]: Wow! Yeah that was definitely not right. I wrote that commit against a different branch and then for some reason it got committed to this branch. Reverted.

[the remark this answers is not in the document]
Comment (Cathyhjj) [reply src/firefly/tests/test_grid_scan_window.py:1]: this is fixed now

File: environment.yml
[the remark this answers is not in the document]
Comment (Cathyhjj) [reply environment.yml:1]: pyarrow==14.0.1 also gives me pydm and qt errors, so I changed back to pyarrow==10.0.1.

**Review by Cathyhjj — APPROVED** [review]

**Pull request merged**